great_expectations/expectations/expectation.py

```diff
@@ -1417,6 +1429,45 @@ def _get_augmented_library_metadata(self):
 
         return AugmentedLibraryMetadata.from_legacy_dict(augmented_library_metadata)
 
+    def _get_maturity_checklist(
+        self,
+        library_metadata: AugmentedLibraryMetadata,
+        description: ExpectationDescriptionDiagnostics,
+        examples: List[ExpectationTestDataCases],
+        tests: List[ExpectationTestDiagnostics],
+        execution_engines: ExpectationExecutionEngineDiagnostics,
+    ) -> ExpectationDiagnosticMaturityMessages:
+        """Generate maturity checklist messages"""
+        concept_only_checks = []
+        experimental_checks = []
+        beta_checks = []
+        production_checks = []
+        checks = []
+
+        checks.append(ExpectationDiagnostics._check_library_metadata(library_metadata))
+        checks.append(ExpectationDiagnostics._check_docstring(description))
+        checks.append(ExpectationDiagnostics._check_example_cases(examples, tests))
+        checks.append(
+            ExpectationDiagnostics._check_core_logic_for_at_least_one_execution_engine(
+                execution_engines
+            )
+        )
+        if library_metadata["maturity"] == "CONCEPT_ONLY":
```

Comment:
Discussed this on a call. Instead of putting all of the checks under a single maturity level, let's split them up, with the proper checks going under each maturity level.

You can see the appropriate levels [here](https://github.com/great-expectations/great_expectations/blob/96423d6b706725f0411450f81cbd58d743e3ad2b/docs/contributing/contributing_maturity.md)